src/convenience/session.ts

```diff
@@ -85,6 +85,30 @@ export interface StorageAdapter<T> {
      * Deletes a value for the given key from the storage.
      */
     delete: (key: string) => MaybePromise<void>;
+    /**
+     * Checks whether a key exists in the storage
+     */
+    has?: (key: string) => MaybePromise<boolean>;
+    /**
+     * Lists all keys
+     */
+    readAllKeys?: () => Iterable<string> | AsyncIterable<string>;
+    /**
+     * Lists all values
+     */
+    readAllValues?: () => Iterable<T> | AsyncIterable<T>;
+    /**
+     * Lists all keys with their values
+     */
+    readAllEntries?: () =>
+        | Iterable<[key: string, value: T]>
+        | AsyncIterable<[key: string, value: T]>;
+    /**
+     * Lists the values for all the keys that match the RegExp pattern
+     * @param pattern RegExp to match keys against
+     * @returns Values with a matching key
+     */
+    readManyValues?: (pattern: RegExp) => Iterable<T> | AsyncIterable<T>;
```

Comment:
@KnorpelSenf what do you think of this? I see it being useful to allow chat-members to read all the members from a given chat

Reply:
Ah I see, ignore the comment below then

Reply:
I think if we need to enumerate based on structural properties of the key, we should rather explicitly encode this in the string. For example, we could roll with a two-level keyed store (like wide-column) or similar approaches.

Reply:
Which regex is it?